### PR TITLE
macro-app-profile

### PR DESCRIPTION
**Pull request by resmaa**
Macros for App Profile

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/957#" title="Author self-approved">resmaa</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
Hi @resmaa. Thanks for your PR.

I'm waiting for a [spectrocloud](https://github.com/orgs/spectrocloud/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/spectrocloud/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: content/docs/04.5-devx/05-app-profile.md
Comment (ravi-k8) [inline content/docs/04.5-devx/05-app-profile.md:71]: We can remove username example

File: content/docs/04.5-devx/05-app-profile.md
Comment (ravi-k8) [inline content/docs/04.5-devx/05-app-profile.md:102]: Replace `Pack` with `App`

File: content/docs/04.5-devx/05-app-profile.md
Comment (ravi-k8) [inline content/docs/04.5-devx/05-app-profile.md:106]: Replace `<variable.name>` with `[VARIABLE_NAME]`

File: content/docs/04.5-devx/05-app-profile.md
Comment (karl-cardenas-coding) [inline content/docs/04.5-devx/05-app-profile.md:101]:
```suggestion
Pack macros are the variables defined in the App Profile and are only resolved at the cluster deployment time.
```
I simplified the statement a bit.

File: content/docs/04.5-devx/05-app-profile.md
Comment (karl-cardenas-coding) [inline content/docs/04.5-devx/05-app-profile.md:73]: We should avoid adding comments in our code snippets as people often copy and paste them. We can move the information provided in the comment to above the code snippet.

File: content/docs/04.5-devx/05-app-profile.md
Comment (karl-cardenas-coding) [inline content/docs/04.5-devx/05-app-profile.md:113]: We might want to consider a table here. If the last variable has a definition, then the rest should also have a definition. That's why a table is helpful in this context, as it will be easier for people to see the variables and their definitions.

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]
I added some suggestions that can help us improve this page.

File: content/docs/04.5-devx/05-app-profile.md
Comment (tiwarisumit3) [inline content/docs/04.5-devx/05-app-profile.md:73]: To refer to output parameters of the below tiers

File: content/docs/04.5-devx/05-app-profile.md
Comment (tiwarisumit3) [inline content/docs/04.5-devx/05-app-profile.md:23]: Mention that in the code editor type {{.spectro.}} to see the list available variables in that particular tier.

File: content/docs/04.5-devx/05-app-profile.md
Comment (tiwarisumit3) [inline content/docs/04.5-devx/05-app-profile.md:21]: Lets mention about two sets of parameters:

1. Input parameters - What user can specify and will be consumed by the current tier. It can even use output variables from below tier.
2. Output parameters - which will be generated once the tier gets deployed. It can be consumed by the above tiers.

File: content/docs/04.5-devx/05-app-profile.md
Comment (resmaa) [reply content/docs/04.5-devx/05-app-profile.md:73]: done

File: content/docs/04.5-devx/05-app-profile.md
Comment (resmaa) [reply content/docs/04.5-devx/05-app-profile.md:23]: image

File: content/docs/04.5-devx/05-app-profile.md
Comment (karl-cardenas-coding) [inline content/docs/04.5-devx/05-app-profile.md:129]: Make this a two columns table with the description in the other columns. Also, if one row has a description, then all other rows should also have a description.

File: content/docs/04.5-devx/05-app-profile.md
Comment (karl-cardenas-coding) [inline content/docs/04.5-devx/05-app-profile.md:118]:
```suggestion
User-provided values that will be consumed by the current tier. It can even use output variables from below tier. This parameter resolves to a value of the variable defined in App Profile tier parameters.yaml file.
```
@resmaa what does this mean `It can even use output variables from below tier`. It's clear what this means.
I also made the sentence more active vs passive by stating, "User-provided values vs what a user can specify". We always want to use direct voice when possible.

File: content/docs/04.5-devx/05-app-profile.md
Comment (karl-cardenas-coding) [inline content/docs/04.5-devx/05-app-profile.md:107]:
```suggestion
* inherited from the lower tiers of the App Profile.
```

File: content/docs/04.5-devx/05-app-profile.md
Comment (karl-cardenas-coding) [inline content/docs/04.5-devx/05-app-profile.md:147]:
```suggestion
The variables are generated once the tier gets deployed. Output variables can be consumed by the higher tiers in the App Profile.
```
Let's be specific of what can consume the output variables.

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]

File: content/docs/04.5-devx/05-app-profile.md
Comment (resmaa) [reply content/docs/04.5-devx/05-app-profile.md]: yes

**Review by tiwarisumit3 — APPROVED** [review]

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/957#" title="Author self-approved">resmaa</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/957#pullrequestreview-1166336762" title="Approved">tiwarisumit3</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by karl-cardenas-coding — APPROVED** [review]

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/957#pullrequestreview-1168783335" title="Approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/957#" title="Author self-approved">resmaa</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/957#pullrequestreview-1166336762" title="Approved">tiwarisumit3</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
